stdlib/math.pyi

```diff
@@ -1,5 +1,9 @@
 import sys
 from typing import Iterable, SupportsFloat, SupportsInt, Tuple, overload
+from typing_extensions import Protocol
+
+class _SupportsTrunc(Protocol):
+    def __trunc__(self) -> int: ...
```

Comment:
Would this be better suited for the local `math` namespace or `_typeshed`?
I don't think it's used outside of `math.trunc`, so it could be a bit too specific for `_typeshed`.

Reply:
I think `_typeshed` is a good fit for these kinds of fundamental protocols, even it not used much.

Reply:
Done as of https://github.com/python/typeshed/pull/6003/commits/19fc5ee0c90d852cfe965db49c343832e58f210e.